src/main/game/entities/factories/UnitFactory.java

```diff
@@ -11,28 +15,64 @@
 import game.entities.units.UnitStats;
 import game.entities.units.UnitType;
 import game.gameboard.Location;
+import game.entities.EntityId;
 
 public class UnitFactory {
     private Map<UnitType, UnitStats> unitStatistics;
 
+    private EntityId entityId;
+    private IdManager colonistIdManager;
+    private IdManager meleeIdManager;
+    private IdManager rangedIdManager;
+    private IdManager explorerIdManager;
+
+    private int newColonistId;
+    private int newMeleeId;
+    private  int newRangedId;
+    private int newExplorerId;
+
     public UnitFactory(){
         this.unitStatistics = new HashMap<UnitType, UnitStats>();
         this.unitStatistics.put(UnitType.COLONIST, new UnitStats(UnitType.COLONIST));
         this.unitStatistics.put(UnitType.EXPLORER, new UnitStats(UnitType.EXPLORER));
         this.unitStatistics.put(UnitType.MELEE, new UnitStats(UnitType.MELEE));
         this.unitStatistics.put(UnitType.RANGED, new UnitStats(UnitType.RANGED));
+        colonistIdManager = new IdManager(0, 9);
+        meleeIdManager = new IdManager(0, 9);
+        rangedIdManager = new IdManager(0, 9);
+        explorerIdManager = new IdManager(0, 9);
     }
 
-    public Unit createUnit(UnitType unit, Location location){
+    public Unit createUnit(UnitType unit, Location location, int playerId){
         switch(unit) {
-            case COLONIST:
-                return new Colonist(unitStatistics.get(unit), location);
-            case EXPLORER:
-                return new Explorer(unitStatistics.get(unit), location);
+            case COLONIST:{
+                try{
+                    newColonistId = this.colonistIdManager.getNewId();
+                    entityId = new EntityId(playerId, EntityTypeEnum.UNIT, EntitySubtypeEnum.COLONIST, newColonistId );
+                    return new Colonist(unitStatistics.get(unit), location, entityId);
+                } catch (IdLimitExceededException e) {}
```

Comment:
Handle these exceptions somehow. Probably want to throw something like ColonistLimitExceeded, ExplorerLimitExceeded, etc. for the caller to handle.